### ISSUE TITLE
Arrow and flatbuffers

### ISSUE THREAD

**Issue by gedw99** [chrislusf/gleam#140]
This would work fantastically with Apache Arrow and flatbuffers.

There is a great golang lib for both now !
This would remove a ton of roadmap work from gleam too.

Flatbuffers golang system for data on disk and marshalling into Google flatbuffers.
https://github.com/urban-wombat

https://www.influxdata.com/blog/influxdata-apache-arrow-go-implementation/

They submitted the golang code to Apache Arrow...
https://github.com/apache/arrow

In summary.. Apache Arrow and flatbuffers provides a streaming data store for gleam to run on top of.
Gotables provides and more rudimentary flat buffers store.
